### PR TITLE
Gen3Index fixes

### PR DESCRIPTION
**Pull request by Avantol13**
correct some issues, ensure params are passed, ensure update calls are successfully constructed

### Bug Fixes
- Ensure params are passed correctly for manifest publishing
- Ensure calls to API are constructed correctly for creating and updating records (`Gen3Index.async_create_record` and `Gen3Index.async_update_record`)
- Fix `backoff` max number of tries override with `GEN3SDK_MAX_RETRIES` environment variable

### CODE REVIEW

File: gen3/index.py [hunk 1]
```diff
@@ -736,15 +738,29 @@ async def async_update_record(
...
+            del record["baseid"]
+            del record["size"]
+            del record["hashes"]
+            del record["did"]
+
+            logging.info(f"PUT-ing record: {record}")
+
             async with session.put(
-                f"{self.client.url}/index/{guid}/rev={revision}",
+                f"{self.client.url}/index/{guid}?rev={revision}",
```
Comment (paulineribeyre) [inline gen3/index.py:763]: oof i can't believe this has been broken for this long

**Review by paulineribeyre — DISMISSED** [review]

File: gen3/index.py (hunk 1 above)
Comment (Avantol13) [reply gen3/index.py:763]: :sweat_smile: me either

File: gen3/index.py [hunk 2]
```diff
@@ -473,30 +473,39 @@ async def async_create_record(
...
                 "hashes": hashes,
                 "size": size,
-                "file_name": file_name,
-                "metadata": metadata,
-                "urls_metadata": urls_metadata,
-                "baseid": baseid,
-                "acl": acl,
-                "authz": authz,
-                "version": version,
+                "urls": urls or [],
```
Comment (BinamB) [inline gen3/index.py:479]: isn't `urls` a required field for indexing? If its not available then it shouldn't be a valid manifest right?

**Review by thanh-nguyen-dang — APPROVED** [review]

File: gen3/index.py (hunk 2 above)
Comment (paulineribeyre) [reply gen3/index.py:479]: it's required to provide a list, but it accepts an empty list, so i think that counts as optional? 😄 this isn't for indexing, it's to index a single record, and there are a lot of use cases for indexing before having the URL

Comment (BinamB) [reply gen3/index.py:479]: gotcha

**Review by BinamB — APPROVED** [review]

**Pull request merged**